### PR TITLE
Fix union type should be printed in the multi-line variant when there are comments

### PR DESCRIPTION
**Pull request by PerfectPan**
## Description

Fix the problem that type should not be considered as hug type when there are comments (#13858)

## Checklist

- [x] I’ve added tests to confirm my change works.
- [x] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by PerfectPan** [comment]
I wonder if I want to contribute to the project like bugfix, it is better to base on next branch or main branch?

File: changelog_unreleased/typescript/13860.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#### Fix union type should be printed in the multi-line variant when there are comments (#13860 by @PerfectPan)
+
+<!-- prettier-ignore -->
+```tsx
+// Input
+type FooBar = 
+  | Number // this documents the first option
+  | void // this documents the second option
+  ;
```
Comment (fisker) [inline changelog_unreleased/typescript/13860.md:9]: Can you add this case to test?

Comment (PerfectPan) [reply changelog_unreleased/typescript/13860.md:9]: OK，I will do it when I get off work today

Comment (fisker) [reply changelog_unreleased/typescript/13860.md:9]: Done. [`dcc74d9` (#13860)](https://github.com/prettier/prettier/pull/13860/commits/dcc74d9459cb80e4a016374918f0daed398dba31)

**Review by fisker — APPROVED** [review]

**Review by thorn0 — APPROVED** [review]

**Pull request merged**